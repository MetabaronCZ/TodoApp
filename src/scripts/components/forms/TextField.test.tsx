import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import { TextField } from 'components/forms/TextField';
import { withMockedProviders } from 'test/component';

describe('components/forms/TextField', () => {
  it('should render text input', () => {
    const { container } = render(
      withMockedProviders(<TextField onChange={() => null} />),
    );
    expect(container).toMatchSnapshot();
    expect(container.querySelector('input')).toBeInTheDocument();
  });

  it('should render textarea', () => {
    const { container } = render(
      withMockedProviders(<TextField textarea onChange={() => null} />),
    );
    expect(container).toMatchSnapshot();
    expect(container.querySelector('textarea')).toBeInTheDocument();
  });

  it('should render props', () => {
    const tree = render(
      withMockedProviders(
        <TextField
          value="Content"
          error="Validation error"
          maxLength={50}
          disabled
          onChange={() => null}
        />,
      ),
    );
    expect(tree.container).toMatchSnapshot();

    const input = tree.container.querySelector('input');
    expect(input).toBeInTheDocument();
    expect(input?.value).toEqual('Content');
    expect(input?.maxLength).toEqual(50);
    expect(input?.disabled).toEqual(true);

    expect(tree.getByText('Validation error')).toBeInTheDocument();
  });

  it('should handle value change', () => {
    const onChange = vi.fn();

    const { container } = render(
      withMockedProviders(<TextField onChange={onChange} />),
    );
    const input = container.querySelector('input');
    expect(input).toBeInTheDocument();

    if (input) {
      fireEvent.change(input, { target: { value: 'Changed' } });
      expect(onChange).toBeCalledTimes(1);
      expect(onChange).toBeCalledWith('Changed');
    }
  });
});
