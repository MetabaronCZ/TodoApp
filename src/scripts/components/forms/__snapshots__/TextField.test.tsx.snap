// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/forms/TextField should render props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 100%;
  padding: 5px 10px;
  background: #fff;
  color: #c22;
  border: none;
  outline: none;
}

.c1:hover:not(:disabled) {
  background-color: #fcfcfc;
}

.c1:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c1:disabled {
  color: #c0c0c0;
}

.c2 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  padding: 0 10px;
  color: #fff;
  background: #c22;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      disabled=""
      maxlength="50"
      type="text"
      value="Content"
    />
    <div
      class="c2"
    >
      Validation error
    </div>
  </div>
</div>
`;

exports[`components/forms/TextField should render text input 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 100%;
  padding: 5px 10px;
  background: #fff;
  border: none;
  outline: none;
}

.c1:hover:not(:disabled) {
  background-color: #fcfcfc;
}

.c1:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c1:disabled {
  color: #c0c0c0;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`components/forms/TextField should render textarea 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 100%;
  padding: 5px 10px;
  background: #fff;
  border: none;
  outline: none;
  min-height: 110px;
  resize: vertical;
}

.c1:hover:not(:disabled) {
  background-color: #fcfcfc;
}

.c1:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c1:disabled {
  color: #c0c0c0;
}

<div>
  <div
    class="c0"
  >
    <textarea
      class="c1"
    />
  </div>
</div>
`;
